andela/ah-fortem-frontend
# 167348690-setup husky

#### Title

Setup Husky
#### Description
Husky is a tool that can help in preventing pushing of code that has things like failing tests, or code with Eslint issues to Github. This will help boost productivity since developers will not have to wait for tools such as hound to sniff around the js files and come up with issues or push code only to wait for `Travis` to report the failing tests. 


#### Type of change







- [x] A chore / Task

#### How Can This Be Tested?

1. Checkout to the `task/167348690-setup-husky` branch & run `npm install`
2. Install this extension in VSCode https://marketplace.visualstudio.com/items?itemName=glen-84.sass-lint
3. Break an eslint rule, like defining a variable and not using it
![image](https://user-images.githubusercontent.com/12128153/61388117-11f58100-a8c0-11e9-961f-ddac059337ae.png)

4. Add a styling in any sass file that does not conform to the sass standards
![image](https://user-images.githubusercontent.com/12128153/61388397-86302480-a8c0-11e9-9308-c364728abbd7.png)
5. Run `git add .` & `git commit`
You will see the following results
![image](https://user-images.githubusercontent.com/12128153/61389051-e9ba5200-a8c0-11e9-9a7f-18f0f252ea35.png)



#### Acceptance Criteria: 

1. Before a successful commit,  `sass styling` & `Eslint` issues should be logged to the console.
2. The staged poorly formated code should be reformated on the fly with prettier
3. The failing tests will be identified

All this information will be known before pushing the code to Github

#### Checklist:

- [x] - install husky, prettier , sass-lint
- [x] - configure hound to use sass-lint
- [x] - setup precommit hook

## More Context

In your `.env` file kindly add the following line `HUSKY_SKIP_HOOKS=1 git rebase ...` this will make husky not run its checks when you are rebasing
If you want to opt-out of the checks you can use `git commit --no-verify`

#### PT stories
[#167348690](https://www.pivotaltracker.com/story/show/167348690)
